### ISSUE TITLE
Clear method disabling new search action 

### ISSUE THREAD

**Issue by BillOTei** [ericgio/react-bootstrap-typeahead#193]
### Version 
1.1.0

### Steps to reproduce
```javascript
<AsyncTypeahead
    options={this.props.communesOrPays}
    onChange={function(e) {
        this.props.onCommuneOrPaysUpdate(e);
        //$('#communeorpays-reset').click();
        //this.inputElement.click();
        if (e.length) {
            /**
             * @property {Typeahead} typeaheadC
             */
            this.refs.typeaheadC.getInstance().clear();
        }
    }.bind(this)}
    onSearch={this.props.searchCommuneOrPays}
    labelKey={this.props.formatCommuneOrPaysLabel}
    disabled={!this.props.personnePhysique.pphDateNaissance}
    ref="typeaheadC"
/>
{/* Can"t reset typeahead without disabling search?!!! */}
{/*<button*/}
    {/*id="communeorpays-reset"*/}
    {/*//ref={function(input) {this.inputElement = input;}.bind(this)}*/}
    {/*className="hidden"*/}
    {/*onClick={function() {this.refs.typeahead.getInstance().clear();}.bind(this)}>*/}
    {/*&nbsp;*/}
{/*</button>*/}
```

### Expected Behavior
first search, selection and then click on an item. 
Input is cleared then when entering new search is triggered.
Tried with a hard-coded button, works when user clicks on it but not when programmaticaly 
clicked (cf code ex)

### Actual Behavior
first search works well, selection and then click on an item. 
Input is cleared but then when entering new search no request/search is triggered.

**Comment by Sailias** [comment]
@BillOTei I have noticed that if you do `setTimeout(this.myClearFunction, 0);` it works

**Comment by hechlerp** [comment]
@Sailias That worked for me, thank you! I'm not sure why, though.

**Comment by NickBrooks** [comment]
@Sailias @hechlerp Same issue for me but workaround also works for me, thanks heaps!

**Comment by ericgio** [comment]
This is happening because of [these lines](https://github.com/ericgio/react-bootstrap-typeahead/blob/master/src/containers/asyncContainer.js#L116-L117) and the async nature of `setState`. `setTimeout` allows the state change to happen before `clear` is called.

Pretty sure the fix is to change those lines to:
```jsx
this.setState({hasSelection: !!selected.length}, () => {
  this.props.onChange && this.props.onChange(selected);
});
```

**Comment by ericgio** [comment]
This is fixed in v2.0.0-rc.1
